### PR TITLE
nixos/buildbot: fix eval, and some twin issues

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test buildbot

File: nixos/modules/services/continuous-integration/buildbot/master.nix [hunk 1]
```diff
@@ -222,7 +222,7 @@ in {
   };
 
   config = mkIf cfg.enable {
-    users.groups = optional (cfg.group == "buildbot") {
+    users.groups = optionalAttrs (cfg.group == "buildbot") {
```
Comment (Ma27) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: wouldn't it be better to always use `mkIf` for subtrees in the module system?

Comment (worldofpeace) [reply nixos/modules/services/continuous-integration/buildbot/master.nix]: They're almost used interchangeably if you search for `users.groups = optionalAttrs` or `users.groups = mkIf` that I lack a preference.

Comment (worldofpeace) [reply nixos/modules/services/continuous-integration/buildbot/master.nix]: Though combinations of the two with `//` is actually a bug, ref https://github.com/NixOS/nixpkgs/pull/72949 and what seemed like a seemingly mysterious issue https://github.com/NixOS/nixpkgs/pull/72916. One should then be mindful to use `mkMerge` and `mkIf` only.

**Pull request merged**